src/platform/plugins/shared/discover/public/application/main/data_fetching/fetch_all.ts

```diff
@@ -123,6 +124,19 @@ export function fetchAll(
         })
       : fetchDocuments(searchSource, params);
     const fetchType = isEsqlQuery ? 'fetchTextBased' : 'fetchDocuments';
+
+    // Build the search request body first (this triggers buildEsQuery)
+    // Then analyze the BUILT ES DSL query
+    // Wrap in try-catch to handle bad query syntax gracefully.
+    let queryAnalysis: MultiMatchAnalysis = { typeCounts: new Map(), rawTypes: [] };
+    try {
+      searchSource.getSearchRequestBody();
+      queryAnalysis = searchSource.getQueryAnalysis();
```

Comment:
It seems odd to me that the consumer of `getQueryAnalysis` should have to be sure that `getSearchRequestBody` is called beforehand. Can we update `getQueryAnalysis` itself to call `getSearchRequestBody` instead?

Reply:
How about I solve it this way: Instead of the current separate code path that records the query body I could piggy-back on the existing request inspector. It already records the request bodies and allows for retrieval afterwards. 

@lukasolson Would you consider that fine or would it abuse the request inspector in your opinion?